include/swift/SIL/OwnershipUtils.h

```diff
@@ -206,6 +206,9 @@ bool isOwnershipForwardingInst(SILInstruction *i);
 /// Is this an instruction that can forward guaranteed ownership.
 bool isGuaranteedForwardingInst(SILInstruction *i);
 
+/// Is this a use that forwards guaranteed ownership.
+bool isGuaranteedForwardingUse(Operand *op);
```

Comment:
@atrick one thing that I think is definitely true (as I have thought about it) is that the notion of "forwarding" should be attached not to a value or an instruction. Instead it should be on the operand (the use) that joins them. That is the point where forwarding happens.

Reply:
Yes ownership applies to an operand but it will generally be useful to group the domain of opcodes into consuming, forwarding, and normal users. It's certainly important for readability. The most clear model will just answer the question: which operations (opcodes) are consuming or forwarding (as opposed to normal uses). For just those operations, which operand is being consumed or forwarded. It will never be more than one. Applies are obviously an exception. There's also the interesting case of "reborrows" for struct/tuple/phi which I think is a separate aspect of the model.